File: internal/observability/sentry.go
package observability

import (
	"log/slog"
	"os"

	"github.com/getsentry/sentry-go"
)

// InitSentry initializes the global Sentry client if SENTRY_DSN is set.
// Returns a shutdown func that flushes pending events.
func InitSentry() func() {
	dsn := os.Getenv("SENTRY_DSN")
	if dsn == "" {
		slog.Info("sentry: DSN not set, skipping initialization")
		return func() {}
	}
	err := sentry.Init(sentry.ClientOptions{
		Dsn:              dsn,
		TracesSampleRate: 0.1,
		Environment:      os.Getenv("SENTRY_ENVIRONMENT"),
		Release:          os.Getenv("AIOJ_RELEASE"),
	})
	if err != nil {
		slog.Error("sentry init failed", "err", err)
		return func() {}
	}
	slog.Info("sentry: initialized", "env", os.Getenv("SENTRY_ENVIRONMENT"))
	return func() { sentry.Flush(2) }
}

File: web/src/lib/sentry.ts
import * as Sentry from '@sentry/react'

export function initSentry() {
    const dsn = import.meta.env.VITE_SENTRY_DSN
    if (!dsn) {
        // Quiet no-op when DSN is unset (dev default).
        return
    }
    Sentry.init({
        dsn,
        environment: import.meta.env.MODE,
        release: import.meta.env.VITE_AIOJ_RELEASE,
        tracesSampleRate: 0.1,
        integrations: [Sentry.browserTracingIntegration()],
    })
}

export { Sentry }
